src/conn/conn_cache_pool.c

```diff
@@ -207,6 +207,11 @@ __wt_cache_pool_config(WT_SESSION_IMPL *session, const char **cfg)
     F_SET(conn, WT_CONN_CACHE_POOL);
 err:
     __wt_spin_unlock(session, &__wt_process.spinlock);
+
+    /*
+     * Cannot use __wt_spin_unlock_if_owned because we can reach this without the lock being
+     * initialized.
+     */
```

Comment:
Can we fix the uninitialized lock?

Reply:
We can't, because there are several places from which we can error out here before we even allocate the struct that contains the lock. But I just realized that we can do this more cleanly by using variable that tracks whether the lock exists (I'll call it `cp_lock_available`). Let me try that.